homeassistant/components/dunehd/config_flow.py

```diff
@@ -0,0 +1,99 @@
+"""Adds config flow for Dune HD integration."""
+import ipaddress
+import logging
+import re
+
+from pdunehd import DuneHDPlayer
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_HOST
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def host_valid(host):
+    """Return True if hostname or IP address is valid."""
+    try:
+        if ipaddress.ip_address(host).version == (4 or 6):
+            return True
+    except ValueError:
+        disallowed = re.compile(r"[^a-zA-Z\d\-]")
```

Comment:
We probably need a helper to validate hostnames.

- They need to be <= 253 chars
- No segment can be all numbers
- They can't start or end with a `-`
- Each label has to be <= 63 chars

Reply:
Found a stackoverflow on this 
https://stackoverflow.com/questions/2532053/validate-a-hostname-string